plugins/modules/public_cloud_instance_private_network.py

```diff
@@ -0,0 +1,199 @@
+    instance_id:
+        required: true
+        description:
+            - The instance id
+    private_network_id:
+        required: true
+        description:
+            - The id of the private_network
+    static_ip:
+        required: false
+        description:
+            - The static IP to set on new interface
+    state:
+        required: false
+        default: present
+        choices: ['present','absent']
+        description: Indicate the desired state of private_network
+"""
+
+EXAMPLES = """
+- name: "Attach a private_network to instance {{ instance_id }} on public cloud OVH"
+  synthesio.ovh.public_cloud_instance_private_network:
+    instance_id: "{{ instance_id }}"
+    service_name: "{{ service_name }}"
+    private_network_id: "{{ private_network_id }}"
+    static_ip: "{{ static_ip }}"
+    state: present
+"""
+
+RETURN = """ # """
+
+
+from ansible_collections.synthesio.ovh.plugins.module_utils.ovh import (
+    ovh_api_connect,
+    ovh_argument_spec,
+)
+
+try:
+    from ovh.exceptions import APIError
+
+    HAS_OVH = True
+except ImportError:
+    HAS_OVH = False
+
+
+def run_module():
+    module_args = ovh_argument_spec()
+    module_args.update(
+        dict(
+            service_name=dict(required=True),
+            state=dict(choices=["present", "absent"], default="present"),
+            instance_id=dict(required=True),
+            private_network_id=dict(required=True),
+            static_ip=dict(required=False, default=None),
+        )
+    )
+
+    module = AnsibleModule(argument_spec=module_args, supports_check_mode=True)
+    client = ovh_api_connect(module)
+
+    service_name = module.params["service_name"]
+    private_network_id = module.params["private_network_id"]
+    static_ip = module.params["static_ip"]
+    instance_id = module.params["instance_id"]
+    state = module.params["state"]
+
+    if module.check_mode:
+        module.exit_json(
+            msg="{}/{} successfully configured ({}) on  private_network {} - (dry run mode)".format(
+                instance_id, service_name, state, private_network_id
+            ),
+            changed=True,
+        )
+
+    is_already_registered = False
+    private_network_if = None
+
+    # list existing interfaces
+    try:
+        interfaces_list = client.get(
+            "/cloud/project/{0}/instance/{1}/interface".format(
+                service_name, instance_id
+            )
+        )
+
+        for netif in interfaces_list:
+            if netif["networkId"] == private_network_id:
+                is_already_registered = True
+                private_network_if = netif
+
+    except APIError as api_error:
+        module.fail_json(msg="Failed to get interfaces list: {0}".format(api_error))
+
+    # Attach or detach
+    if state == "present":
+        if not is_already_registered:
+            try:
+                if static_ip:
```

Comment:
I'm not sure about that, but the API seems to accept a null value or an empty string for the `ip` property.
You could remove this if/else and always send this `ip` property